metadata-ingestion/tests/unit/sdk_v2/test_lineage_client.py

```diff
@@ -561,3 +562,261 @@ def test_add_datajob_lineage_validation(client: DataHubClient) -> None:
+
+    # Call the method
+    client.lineage.add_lineage(upstream=upstream, downstream=downstream)
+
+    # Validate lineage MCPs against golden file
+    assert_client_golden(
+        client, _GOLDEN_DIR / "test_lineage_datajob_to_dataset_golden.json"
+    )
+
+
+def test_add_lineage_datajob_to_datajob(client: DataHubClient) -> None:
+    """Test add_lineage method with datajob to datajob."""
+    upstream = (
+        "urn:li:dataJob:(urn:li:dataFlow:(airflow,example_dag,PROD),upstream_job)"
+    )
+    downstream = (
+        "urn:li:dataJob:(urn:li:dataFlow:(airflow,example_dag,PROD),downstream_job)"
+    )
+
+    # Call the method
+    client.lineage.add_lineage(upstream=upstream, downstream=downstream)
+
+    # Validate lineage MCPs against golden file
+    assert_client_golden(
+        client, _GOLDEN_DIR / "test_lineage_datajob_to_datajob_golden.json"
+    )
+
+
+def test_add_lineage_invalid_upstream(client: DataHubClient) -> None:
+    """Test add_lineage method with invalid upstream URN."""
+    upstream = "urn:li:glossaryNode:something"
+    downstream = "urn:li:dataset:(urn:li:dataPlatform:snowflake,target_table,PROD)"
+
+    # Should raise an InvalidUrnError
+    with pytest.raises(
+        SdkUsageError,
+        match="Unsupported entity type combination: glossaryNode -> dataset",
+    ):
+        client.lineage.add_lineage(upstream=upstream, downstream=downstream)
+
+
+def test_add_lineage_invalid_downstream(client: DataHubClient) -> None:
+    """Test add_lineage method with invalid downstream URN."""
+    upstream = "urn:li:dataset:(urn:li:dataPlatform:snowflake,source_table,PROD)"
+    downstream = "urn:li:glossaryNode:something"
+
+    # Should raise an InvalidUrnError
+    with pytest.raises(
+        SdkUsageError,
+        match="Unsupported entity type combination: dataset -> glossaryNode",
+    ):
+        client.lineage.add_lineage(upstream=upstream, downstream=downstream)
+
+
+def test_add_lineage_invalid_parameter_combinations(client: DataHubClient) -> None:
+    """Test add_lineage method with invalid parameter combinations."""
+    # Dataset to DataJob with column_lineage (not supported)
+    with pytest.raises(SdkUsageError):
+        client.lineage.add_lineage(
+            upstream="urn:li:dataset:(urn:li:dataPlatform:snowflake,source_table,PROD)",
+            downstream="urn:li:dataJob:(urn:li:dataFlow:(airflow,example_dag,PROD),process_job)",
+            column_lineage={"target_col": ["source_col"]},
+        )
+
+    # Dataset to DataJob with query_text (not supported)
+    with pytest.raises(SdkUsageError):
+        client.lineage.add_lineage(
+            upstream="urn:li:dataset:(urn:li:dataPlatform:snowflake,source_table,PROD)",
+            downstream="urn:li:dataJob:(urn:li:dataFlow:(airflow,example_dag,PROD),process_job)",
+            query_text="SELECT * FROM source_table",
+        )
+
+    # DataJob to Dataset with column_lineage (not supported)
+    with pytest.raises(SdkUsageError):
```

Comment:
with pytest.raises(), also provide a regex to check the error message says something about "unsupported lineage"